ymdflag.go

```diff
@@ -4,139 +4,225 @@ package ymdflag
 
 import (
 	"fmt"
-	"os"
 	"strconv"
 	"time"
 	"unicode"
 )
 
 // YMDFlag represents a Golang flag.Value for `YYYYMMDD`-specified dates.
+//
+// To faciliates the use of YMD dates in command line flags, it implements the
+// [flag.Value interface], making it compatible with the [flag] and [pflag] packages.
+//
+// It stores an integral `yyyymmddd`.  The special value of 0 indicates that the value
+// is indeterminate and may be may be auto-populated by `UpdateNilToNow`, `AsTime`, or `AsTimeWithLoc`.
+//
+// [flag.Value interface]: https://pkg.go.dev/flag#Value
+// [flag]: https://pkg.go.dev/flag
+// [pflag]: https://pkg.go.dev/github.com/spf13/pflag
 type YMDFlag struct {
 	yyyymmdd int // internal yyyymmdd value, nil values might be mutated
 }
 
+///////////////////////////////////////////////////////////////////////////////
+
+// TODO: internal error consts how?
+
+// YMDtoTime returns the Time corresponding to the YYYYMMDD in the specified location, without validating the argument.`
+// A value of 0 returns a Zero Time, independent of location.
+// A nil location implies local time.
+func YMDToTime(yyyymmdd int, loc *time.Location) time.Time {
+	if yyyymmdd == 0 {
+		return time.Time{}
+	}
+	var year int = yyyymmdd / 10000
+	var month int = (yyyymmdd % 10000) / 100
+	var day int = yyyymmdd % 100
+	if loc == nil {
+		loc = time.Local
+	}
+	return time.Date(year, time.Month(month), day, 0, 0, 0, 0, loc)
+}
+
+// TimeToYMD returns the YYYYMMDD for the Time in its location.
+// A zero time returns a 0 value.
+func TimeToYMD(t time.Time) int {
+	if t.IsZero() {
+		return 0
+	} else {
+		return 10000*t.Year() + 100*int(t.Month()) + t.Day()
+	}
+}
```

Comment:
The date returned will be the time.Time's date which may be different than the local date. I guess this is fine but I worry may be unexpected.